src/main/java/organice/logic/commands/DoneCommand.java

```diff
@@ -0,0 +1,92 @@
+package organice.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static organice.logic.commands.MatchCommand.match;
+
+import organice.model.Model;
+import organice.model.person.Donor;
+import organice.model.person.Nric;
+import organice.model.person.Patient;
+import organice.model.person.exceptions.PersonNotFoundException;
+
+public class DoneCommand extends Command {
+    public static final String COMMAND_WORD = "done";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finish the processing of patients and donors"
+            + "Parameters: ic/PATIENT NRIC ic/DONOR NRIC \n"
+            + "Example: " + COMMAND_WORD + " ic/s4512345A ic/s7711123C";
+
+    public static final String MESSAGE_SUCCESS = "Done processing the patient and donor";
+    public static final String MESSAGE_NOT_PROCESSED = "Donor or patient NRIC must be valid";
+
+    private String firstNricString;
+    private String secondNricString;
+
+    private Nric firstNric;
+    private Nric secondNric;
+
+    private Donor donor;
+    private Patient patient;
+
+    private Nric patientNric;
+    private Nric donorNric;
+
+
+    public DoneCommand(String firstNricString, String secondNricString) {
+        requireNonNull(firstNricString, secondNricString);
+        firstNric = new Nric(firstNricString);
+        secondNric = new Nric(secondNricString);
+    }
+
+    public boolean isValidDonorPatientPair(Nric firstNric, Nric secondNric, Model model) {
+        if (model.hasDonor(firstNric)) {
+            donorNric = firstNric;
+            donor = model.getDonor(donorNric);
+
+            patientNric = secondNric;
+            patient = model.getPatient(patientNric);
+        } else {
+            patientNric = firstNric;
+            patient = model.getPatient(patientNric);
+
+            donorNric = secondNric;
+            donor = model.getDonor(donorNric);
+        }
+        if (model.hasPatient(patientNric) && model.hasDonor(donorNric)
+                && match(donor, patient))  {
+            //todo check if tasklist isempty
+            return true;
+        } else {
+            return false;
+        }
+
+    }
+
+    @Override
+    public CommandResult execute(Model model) {
+        requireNonNull(model);
+        try {
+            if (isValidDonorPatientPair(firstNric, secondNric, model)) {
+                model.getFilteredPersonList();
+
+                donor.setStatus("done");
+                patient.setStatus("done");
+
+                model.deletePerson(donor);
```

Comment:
Any reason why you set the status as "done", then delete the donor and patient off the list?

Reply:
This is the minimal implementation. Becuase at first we said we will delete the patient off the list. In fiture for development, we will see the result of the cross match and then set the status to done or processing again depending on the result. Then, the patients and donors will be archive like hidden from the model with the status done. Smt like that

Reply:
Noted